### ISSUE TITLE
Resolvi differential expression takes long time to run

### ISSUE THREAD

**Issue by jairaj-mathur** [scverse/scvi-tools#3672]
Differential expression with group by as leiden_resolvi (one vs none for all Leiden clusters) takes 30 minutes per cluster. Even if run for a single comparison (one vs all) it takes 30 minutes.

My dataset is Xenium 5k, 200k cells. Is this normal?

```python
# Your code here
```

```pytb
[Paste the error output produced by the above code here]
```

#### Versions:

1.4.1

**Comment by ori-kron-wis** [comment]
Please post your code here so we can review it.
How many clusters do you have? 
Did you try with batch_correction=False (like I suggested in discourse)

**Comment by jairaj-mathur** [comment]
Hi Ori, it was ran with 

```
RESOLVI.setup_anndata(concat_adata, categorical_covariate_keys=["region"], batch= None, layer= "counts")
model= RESOLVI(concat_adata, semisupervised=false downsample_counts=false)
model.train(batch_size=256, max_epochs=100)
adata= model.adata.copy()
adata.obsm[X_resolvi] = model.get_latent_representation()
sc.pp.neighbors(adata, use_rep="X_resolvi")
sc.pl.umap(adata)
sc.tl.leiden(adata, resolution=0.75, flavor="igraph", niter=-1)
results=model.differential_expression(adata, groupby=leiden)
```

This yielded 22 clusters, and took 9.5 hours

From your discourse post, I believed that batch_correction is set to False by default , thus didn't run. I can try again and let you know.

**Comment by canergen** [comment]
I'm sorry for this issue. The likely reason is that importance sampling in pyro can be slow. Apparently, currently importance sampling can't be disabled: https://github.com/scverse/scvi-tools/blob/14daf4729b9ce36102eed964d8bc3adc1c1223bc/src/scvi/external/resolvi/_model.py#L471. We should fix this and add an option to use get_normalized_expression instead for a faster computation.

**Comment by jairaj-mathur** [comment]
No worries, thank you for your time to get into the root of this. I will close this issue for now.

**Comment by canergen** [comment]
Reopened as I would argue we should harmonize behavior to other models.

**Comment by ori-kron-wis** [comment]
@jairaj-mathur, please see this branch: https://github.com/scverse/scvi-tools/tree/Ori-3672-Resolvi-DE-IS-flag
It consists of 2 new parameters for the RESOLVI DE, n_samples and run_IS_DE. They will both control the speed (but also accuracy) of the DE.
Try to run the DE from this branch with `run_IS_DE=False` and a lower number of n_samples (1-3), which should improve speed drastically. 
Can you confirm it does?

**Comment by ori-kron-wis** [comment]
Made a revision to this issue correction that can be found here: https://github.com/scverse/scvi-tools/pull/3708

It might be that we need to add a fix to other model sampling behavior, but this one deals specifically with the pyro IS function, which ResolVI is the only model to be using, so let's concentrate only on this here.
